src/clock.js

```diff
@@ -0,0 +1,21 @@
+class Clock {
+  // Return the current time in milliseconds
+  msec () {
+    return new Date().getTime();
+  }
+
+  // Run fn periodically, whether async or not, surviving errors
+  periodically (interval, name, fn) {
+    let iterator = () => {
```

Comment:
Thanks.  That synchronous-exceptions issue has bitten me in Python, too, so I should have known better!